app/services/application_state.rb

```diff
@@ -0,0 +1,38 @@
+class ApplicationState
```

Comment:
This comment is related to the previous one.
I find the name a bit misleading since it does not represent just the state but the whole workflow,
what about `ApplicationWorkflow`, `ApplicationFlow`, `ApplicationLogic`,  `ApplicationBusinessLogic`, ... I don't know :)

Reply:
I'm wary of `ApplicationState` as `Application` doesn't really exist in our code. How about `ApplicationChoiceStatusChange`? That is its job.